src/pluginplay/printing/mermaid.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module/module_class.hpp"
+#include "module_manager/module_manager_class.hpp"
+#include <pluginplay/pluginplay.hpp>
+#include <sstream>
+
+void add_letter(std::string& code) {
+    for(int i = 2; i >= 0; --i) {
+        if(code[i] == 'Z') {
+            code[i] = 'A';
+        } else {
+            code[i]++;
+            return;
+        }
+    }
+}
```

Comment:
```suggestion
void add_letter(std::string& code) {
    assert(code.size() == 2);
    for(int i = 2; i >= 0; --i) {
        if(code[i] == 'Z') {
            code[i] = 'A';
        } else {
            code[i]++;
            return;
        }
    }
    throw std::runtime_error("String is maxed out");
}
```

You should always check your assumptions (like the size of `code` and that you  actually updated the string, i.e., that you hit the else condition).